From: Gerald Schaefer <geraldsc@de.ibm.com>
Subject: IBM Codestream linux-2.6.15 october2005, patch 02-42

  Description: qeth: setting of attribute "route6" to "primary_router"
               works only once.
  Symptom:     set route6 attribute to primary_router and then back to no_router.
               when trying to set primary_router again qeth reports a failure
               message in dmesg and does not set route6 to primary_router.
  Problem:     when setting route6 attribute back to no_router qeth does not
               issue an IP ASSIST command to reset router value to no_router.
               Once primary_router is set device stays in this mode.
  Solution:    Issue an IP ASSIST command when no_router is set in route6.
               Device will be reset and thus will not longer run as a primary
               router.
  Problem-ID:  -
  Archs:       s390-31, s390-64

Acked-by: Hannes Reinecke <hare@suse.de>

--- linux-2.5/drivers/s390/net/qeth_main.c	19 Mar 2006 19:50:37 -0000	1.252.2.5
+++ linux-2.5/drivers/s390/net/qeth_main.c	19 Mar 2006 20:00:16 -0000	1.252.2.6
@@ -7344,11 +7344,6 @@
 	qeth_correct_routing_type(card, &card->options.route6.type,
 				  QETH_PROT_IPV6);
 
-	if ((card->options.route6.type == NO_ROUTER) ||
-	    ((card->info.type == QETH_CARD_TYPE_OSAE) &&
-	     (card->options.route6.type == MULTICAST_ROUTER) &&
-	     !qeth_is_supported6(card,IPA_OSA_MC_ROUTER)))
-		return 0;
 	rc = qeth_send_setrouting(card, card->options.route6.type,
 				  QETH_PROT_IPV6);
 	if (rc) {

